lib/router.js

```diff
@@ -339,41 +339,40 @@ Router.prototype._handleSwaggerPathSpec = function(node, pathspec, scope) {
 
     // Load sub-spec
     var loaderPromise = P.resolve();
-    // XXX: find a better way to detect this?
-    if (pathspec.swagger) {
-        if (pathspec.info && pathspec.info['x-is-api-root']) {
-            // This is a new API at a path like /en.wikipedia.org/v1, so create a
-            // new specRoot.
-            var specRoot = Object.assign({}, pathspec);
-            specRoot.paths = {};
-            specRoot.definitions = {};
-            specRoot.securityDefinitions = {};
-            specRoot['x-default-params'] = {};
-            specRoot.basePath = (scope.specRoot.basePath || '') + scope.prefixPath;
-            // XXX: The basePath is incorrect when shared between domains. Set
-            // it dynamically for each request instead?
-            specRoot.basePath = scope.prefixPath;
-            scope = scope.makeChild({
-                specRoot: specRoot,
-                operations: {},
-                prefixPath: '',
-            });
+    if (parentSegment && parentSegment.name === 'api') {
+        // This is a new API at a path like /en.wikipedia.org/v1, so create a new specRoot.
+        var specRoot = Object.assign({}, pathspec);
+        specRoot.swagger = specRoot.swagger || '2.0';
+        specRoot.paths = {};
+        specRoot.definitions = {};
+        specRoot.securityDefinitions = {};
+        specRoot['x-default-params'] = {};
+        specRoot.basePath = (scope.specRoot.basePath || '') + scope.prefixPath;
+        // XXX: The basePath is incorrect when shared between domains. Set
+        // it dynamically for each request instead?
+        specRoot.basePath = scope.prefixPath;
+        scope = scope.makeChild({
+            specRoot: specRoot,
+            operations: {},
+            prefixPath: '',
+        });
 
-            var listNode = new Node();
-            listNode.value = {
-                specRoot: specRoot,
-                methods: {},
-                path: specRoot.basePath + '/',
-                globals: node.value.globals,
-            };
-            node.setChild('', listNode);
-            loaderPromise = loaderPromise.then(function() {
-                return self._handleSwaggerSpec(node, pathspec, scope);
-            });
-        } else {
-            // We are loading a sub-spec.
-            loaderPromise = self._handleSwaggerSpec(node, pathspec, scope);
-        }
+        var listNode = new Node();
+        listNode.value = {
+            specRoot: specRoot,
+            methods: {},
+            path: specRoot.basePath + '/',
+            globals: node.value.globals,
+        };
+        node.setChild('', listNode);
+
+        loaderPromise = loaderPromise.then(function() {
+            return self._handleSwaggerSpec(node, pathspec, scope);
+        });
+    } else {
```

Comment:
@gwicke I think you mean `securityDefinitions` stanzas, not `security` stanzas?

For secDef, having them in non-root spec doesn't make any sense since the AuthService looks for the them only in the rootSpec for the current request chain. 

Also, according to the swagger spec schema, the `securityDefinitions` and `definitions` properties can appear only at the top-level, which is represented by a rootSpec in our code. This means, that actually `handleSwaggerSpec` should be called only for the `rootSpec`, which beginning is now defined by an `api` prefix. 

So, I think that the current change is correct, and actually, we need to remove several more `handleSwaggerSpec` calls, but right now it's too risky, as we don't have enough tests for different kinds of specs. I'm currently writing those tests, but they depend on this change.